mathics/builtin/matrices/constrmatrix.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+from mathics.builtin.base import Builtin
+from mathics.core.expression import Expression
+from mathics.core.atoms import Integer0
+
+
+class DiagonalMatrix(Builtin):
+    """
+    <dl>
+      <dt>'DiagonalMatrix[$list$]'
+      <dd>gives a matrix with the values in $list$ on its diagonal and zeroes elsewhere.
+    </dl>
+
+    >> DiagonalMatrix[{1, 2, 3}]
+     = {{1, 0, 0}, {0, 2, 0}, {0, 0, 3}}
+    >> MatrixForm[%]
+     = 1   0   0
+     .
+     . 0   2   0
+     .
+     . 0   0   3
+
+    #> DiagonalMatrix[a + b]
+     = DiagonalMatrix[a + b]
+    """
+
+    summary_text = "Return a diagonal matrix with the elements of a given list"
```

Comment:
A small thing that I also noticed on the last PR. Summary text doesn't start out with a capital letter. You'll find mention and discussion of this in another draft PR. 

Also other than the two summary texts you've written I can't think of any other one that starts off "returns a".  Diagonal start out "gives a".  

One last small thing is that we want to eliminate the use of "#>" (a test that is run but not shown in documentation). These were added before we had proper pytests for such things that can be found in the test directory. 

You attention to detail is appreciated.

Reply:
Ok. Regarding "#>", should I remove the whole line from tensor.py too?

Reply:
Yes - if we could add these as pytests in the `test` directory that would be awesome. 

As mentioned before there is a lot of cleanup work that this project needs because it was greatly expanded before some basic problems were addressed.